storage/watchable_store.go

```diff
@@ -279,10 +285,14 @@ func (s *watchableStore) syncWatchers() {
 			continue
 		}
 		if eb.moreRev != 0 {
+			w.mu.Lock()
 			w.cur = eb.moreRev
+			w.mu.Unlock()
 			continue
 		}
+		w.mu.Lock()
 		w.cur = curRev
```

Comment:
@xiang90 I meant here from the comment. https://github.com/coreos/etcd/pull/4661/files#r54833601